test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/Helpers/ProcessHelperTests.cs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace TestPlatform.CrossPlatEngine.UnitTests
+{
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using Moq;
+    using System.Diagnostics;
+    using Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client;
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Engine;
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using System;
+    using System.Reflection;
+    using System.Linq;
+    using Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Helpers;
+    using Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting;
+
+    [TestClass]
+    public class ProcessHelperTests
+    {
+        private readonly ProxyOperationManager testOperationManager;
+
+        private readonly TestableTestHostManager testHostManager;
+
+        private readonly Mock<ITestRequestSender> mockRequestSender;
+
+        private TestableProcessHelper processHelper;
+
+        private int connectionTimeout = 400;
+
+        private int errorLength = 20;
+
+        public ProcessHelperTests()
+        {
+            this.mockRequestSender = new Mock<ITestRequestSender>();
+            this.mockRequestSender.Setup(rs => rs.WaitForRequestHandlerConnection(this.connectionTimeout)).Returns(true);
+
+            this.processHelper = new TestableProcessHelper();
+            this.testHostManager = new TestableTestHostManager(Architecture.X64, Framework.DefaultFramework, this.processHelper, true);
+
+            this.testOperationManager = new TestableProxyOperationManager(this.mockRequestSender.Object, this.testHostManager, this.connectionTimeout, this.errorLength);
+        }
+
+        [TestMethod]
+        public void ErrorMessageShouldBeReadAsynchronously()
+        {
+            string errorData = "Custom Error Strings";
+            this.processHelper.SetErrorMessage(errorData);
+
+            this.mockRequestSender.Setup(rs => rs.InitializeCommunication()).Returns(123);
+
+            this.testOperationManager.SetupChannel(Enumerable.Empty<string>());
+
+            Assert.AreEqual(this.testOperationManager.GetStandardError(), errorData);
+        }
+
+        [TestMethod]
+        public void ErrorMessageShouldBeTruncatedToMatchErrorLength()
+        {
+            string errorData = "Long Custom Error Strings";
```

Comment:
Suggest to organize unit tests as follows for readability (comments are not required, newlines are only between blocks):
```
// Block of arrange
int x = 10;
double y = 9.0

// Block of act
CallFoo(y);

// Block of assert
Assert(12.0, y);
Assert(29, x);
```